packages/itwin/grouping-mapping-widget/src/widget/components/Grouping.tsx

```diff
@@ -383,35 +410,143 @@ export const Groupings = ({ mapping, goBack }: GroupsTreeProps) => {
+                onClick={addGroup}
+              >
+                {isLoadingQuery ? "Loading Group(s)" : "Add Group"}
+              </Button>
+              <ButtonGroup className="toolbar-buttons">
+                <ToggleSwitch
+                  label="Color by Group"
+                  labelPosition="left"
+                  className="group-view-icon toggle"
+                  disabled={isLoadingQuery}
+                  checked={showGroupColor}
+                  onChange={toggleGroupColor}
+                ></ToggleSwitch>
+                <IconButton
+                  title="Show All"
+                  disabled={isLoadingQuery}
+                  styleType="borderless"
+                  className="group-view-icon"
+                >
+                  <SvgVisibilityShow onClick={showAll}></SvgVisibilityShow>
+                </IconButton>
+                <IconButton
+                  title="Hide All"
+                  disabled={isLoadingQuery}
+                  styleType="borderless"
+                  className="group-view-icon"
+                >
+                  <SvgVisibilityHide onClick={hideAll}></SvgVisibilityHide>
+                </IconButton>
+              </ButtonGroup>
+            </div>
+            {isLoading ? (
+              <LoadingOverlay />
+            ) : groups.length === 0 ? (
+              <EmptyMessage message="No Groups available." />
+            ) : (
+              <div className="group-list">
+                {groups.map((g) => (
+                  <GroupTile
+                    key={g.id}
+                    title={g.groupName ? g.groupName : "Untitled"}
+                    subText={g.description}
+                    actionGroup={
+                      <div className="actions">
+                        <IconButton
+                          disabled={isLoadingQuery}
+                          styleType="borderless"
+                          className="group-view-icon"
+                        >
+                          {g.id && hiddenGroupsIds.includes(g.id) ? (
+                            <SvgVisibilityHide
+                              onClick={async () => {
+                                await showGroup(g);
+                                setHiddenGroupsIds(
+                                  hiddenGroupsIds.filter((id) => g.id !== id),
+                                );
+                              }}
+                            ></SvgVisibilityHide>
+                          ) : (
+                            <SvgVisibilityShow
+                              onClick={async () => {
+                                await hideGroups([g]);
+                                setHiddenGroupsIds(
+                                  hiddenGroupsIds.concat(g.id ? [g.id] : []),
+                                );
+                              }}
+                            ></SvgVisibilityShow>
+                          )}
+                        </IconButton>
```

Comment:
Oh I just found out why the buttons felt unresponsive. You should wrap the SVG's individually with the IconButton component and move the onClick callback on those. The onClick was only triggering when you would press on the SVG itself not the button square.  
```
<IconButton
	disabled={isLoadingQuery}
	styleType="borderless"
	className="group-view-icon"
	onClick={async () => {
	  await showGroup(g);
	  setHiddenGroupsIds(
		hiddenGroupsIds.filter((id) => g.id !== id),
	  );
	}}
	>
	<SvgVisibilityHide />
</IconButton>
```

Should fix the rest of them as well.